readthedocs/proxito/views/serve.py

```diff
@@ -89,6 +89,24 @@ def get(self,
         ``subproject_slash`` is used to determine if the subproject URL has a slash,
         so that we can decide if we need to serve docs or add a /.
         """
+        unresolved_domain = request.unresolved_domain
+        # Handle requests that need canonicalizing,
+        # e.g. HTTP -> HTTPS, redirect to canonical domain, etc.
```

Comment:
```suggestion
        # Handle requests that need canonicalizing first,
        # e.g. HTTP -> HTTPS, redirect to canonical domain, etc.
        # We run this here to save any logic for easily cached responses.
```

Reply:
I do want to note here again my comment around multiple HTTP requests being _very slow_ for the end user, but more efficient for us. 

https://github.com/readthedocs/readthedocs.org/issues/10065#issuecomment-1447173681

Probably worth mentioning this in some fashion here, if we're explicitly making that trade off, also.